extensions/guacamole-auth-cas/src/main/java/org/apache/guacamole/auth/cas/util/TokensAndGroups.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.guacamole.auth.cas.util;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.HashSet;
+
+    /**
+     * A class to facilitate the return values of both Tokens and Groups
+     *
+     * @param tokens, groups
+     *     A Map<String, String> of tokens, parsed from attributes returned
+     *     by CAS.
+     *     A Set<String> of groups, parsed from attributes returned by CAS.
+     *
+     * @return an instance of itself
+     *
+     */
+    public class TokensAndGroups {
+        public final Map<String, String> tokens;
+        public final Set<String> effectiveGroups;
+        public <TokensAndGroups> TokensAndGroups(Map<String, String> tokens, Set<String> effectiveGroups) {
```

Comment:
Why is this constructor declared as if it leverages generics (`<TokensAndGroups>`)?

Reply:
lack of knowing any better.   This is my first attempt at any java whatsoever, so I'm an admitted rank amateur.   How should I do it?

Reply:
Just `public TokensAndGroups(...` without any type specification.  Since the name of the method is the same as the name of the class, Java knows it is a constructor and that it is returning that as its type.